### PR TITLE
Re-factor for Iteration 2

### PR DESCRIPTION
**Pull request by javier-aguilar**
Now that all the methods are in game-collection branch, I went and made sure that they all pass the spec harness. (They do)

Additionally, I made sure that they all have tests written for each method. I also removed one method in game class that was unused (total score) and was the exact same as another (total score and total goals)

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -3,69 +3,52 @@
...
-  def average_goals_per_season
-    @games.reduce(Hash.new(0)) do |acc, game|
-      acc[game.season] += (game.total_goals / count_of_games_by_season[game.season].to_f).round(2)
-      acc
-    end
+  def average_goals_by_season
+    @game_stats.average_goals_by_season
   end
+
 end
```
Comment (steve-meyers) [inline lib/stat_tracker.rb:54]: I like that all these methods have been moved to game_collections.

File: test/stat_tracker_test.rb [hunk 2]
```diff
@@ -56,7 +61,14 @@ def test_average_goals_per_game
...
+      "20122013" => 4.4,
+      "20172018" => 4,
+      "20162017" => 5,
+      "20152016" => 7,
+      "20132014" => 5
+    }
+    assert_equal expected_hash, @stat_tracker.average_goals_by_season
   end
+
 end
```
Comment (steve-meyers) [inline test/stat_tracker_test.rb:74]: The formatting for these hashes in the test looks better.

**Review by steve-meyers — APPROVED** [review]
This looks like a clean version of iteration 2 and is ready to merge to master.

**Comment by mlynch5187** [comment]
Great refactoring!

**Pull request merged**